delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resources/ResourcesImpl.scala

```diff
@@ -109,7 +109,7 @@ final class ResourcesImpl private (
       schemaOpt: Option[IdSegment]
   ): IO[ResourceRejection, Option[DataResource]] =
     (for {
-      project      <- fetchProject(projectRef)
+      project      <- projects.fetchFromCache(projectRef)
```

Comment:
the entries in the cache may come later; either fallback on fetch or just use fetch (fetch should be quite performant IMHO)